python/cudf/cudf/core/column/datetime.py

```diff
@@ -269,6 +270,21 @@ def __contains__(self, item: ScalarLike) -> bool:
             "cudf.core.column.NumericalColumn", self.astype("int64")
         )
 
+    def _validate_fillna_value(
+        self, fill_value: ScalarLike | ColumnLike
+    ) -> plc.Scalar | ColumnBase:
+        """Align fill_value for .fillna based on column type."""
+        if (
+            isinstance(fill_value, np.datetime64)
+            and self.time_unit != np.datetime_data(fill_value)[0]
+        ):
+            # TODO: Disallow this cast
```

Comment:
I feel like a lot of your PRs have had these kinds of comments. Do they all fall into similar buckets? Should we open some issues for tracking?

Reply:
Actually I double checked pandas and our casts here matches the pandas behavior for this method (although I'm not fond of it)

Reply:
The question about tracking TODOs still applies but I'm good with holding off on that. grepping the codebase for these small things is OK with me for now given how much we're churning internally anyway.